Compilation fixes

jsign opened the pull request:
This PR makes some fixes due to Zig breaking changes.

cc @gballet

jsign committed:
fixes

Signed-off-by: Ignacio Hagopian <jsign.uy@gmail.com>

gballet commented on src/main.zig:
```diff
@@ -268,6 +268,7 @@ const RLPEncodablePerson = struct {
     age: u8,
 
     pub fn encodeToRLP(self: RLPEncodablePerson, list: *ArrayList(u8)) !void {
+        _ = self;
```
why not renaming `self` as `_` here?

jsign replied:
This is the default way the compiler fmt autofixes unused variables (since that isn't allowed).
So.. just left what the format decided.

gballet reviewed (commented):
LGTM. I left a comment for the last change. The thing is, I just need that last change to compile with `0.10.0` so, ideally, we'd be able to check the compiler version and take the version that depends on the compiler (I'm talking about the `field_type` -> `type` change) so that we support the entire `0.10` line. But no need to do that right now, it's still an experimental lib. I'll upgrade my compiler.

gballet commented:
Just added a README to warn about the compiler version.

gballet commented:
I also added a CI script with the correct zig version, could you please rebase so that we can be sure that the tests pass?

jsign commented:
> LGTM. I left a comment for the last change. The thing is, I just need that last change to compile with `0.10.0` so, ideally, we'd be able to check the compiler version and take the version that depends on the compiler (I'm talking about the `field_type` -> `type` change) so that we support the entire `0.10` line. But no need to do that right now, it's still an experimental lib. I'll upgrade my compiler.

Yeah, my target is always getting up to date with the latest `master` in https://ziglang.org/download/.
That is the latest stable master that I think now is update daily or similar.

The way I kee up to date with this is with `snap refresh --edge zig`, so I run that once in a while and it automatically updates.
Just mentioning if that helps with the burden.

I'll rebase and ping.

jsign commented:
Meh, I've to fix some extra thing about `build.zig` since tests won't run because of other breaking changes in the Builder API.
I'll do it later and fix that too.

jsign committed:
fix build and ci

Signed-off-by: Ignacio Hagopian <jsign.uy@gmail.com>

jsign commented:
@gballet, ready for another look. 
Basically I rebased it and [had to fix some extra things](https://github.com/gballet/zig-rlp/pull/1/commits/a61623363efede41d7732e46ed412752d4ee0a4a) in `build.zig`. Now doing `zig build test` runs fine in my machine. 

I also updated Zig version to master since the only reasonable thing to do is try to be on the tip-ish.
As I mentioned, the easiest way is to install Zig via snap and do `snap refresh --edge zig` once in a while and be up to date.

We'll see when you approve the workflow run if things run fine.

gballet commented:
> @gballet, ready for another look. Basically I rebased it and [had to fix some extra things](https://github.com/gballet/zig-rlp/pull/1/commits/a61623363efede41d7732e46ed412752d4ee0a4a) in `build.zig`. Now doing `zig build test` runs fine in my machine.
> 
> I also updated Zig version to master since the only reasonable thing to do is try to be on the tip-ish. 

Yeah not sure how I feel about that, because a lot of people like me are using the last stable release. Admittedly, it doesn't matter right now, as I assume no one else is using this lib... but it would make sense to support at least the latest stable release. I'm aware that zig evolves a lot right now, so that might be too much of an ask for now, though.

> As I mentioned, the easiest way is to install Zig via snap and do `snap refresh --edge zig` once in a while and be up to date.

That's if your platform supports snap, which some of mines do not.

> We'll see when you approve the workflow run if things run fine.

Didn't expect you not to be able to run it, but it makes sense since I haven't invited you to collaborate.

gballet commented on .github/workflows/ci.yml:
```diff
@@ -16,7 +16,7 @@ jobs:
       uses: korandoru/setup-zig@v1
       with:
-        zig-version: 0.10.1
+        zig-version: master
```
This really isn't great, because zig changes a lot and so from one CI run to the next, we could end up with something that worked and no longer does, which would be confusing as the zig version isn't displayed. I'll make sure it does, so that we can live with this for the time being.

gballet reviewed (approved):
LGTM. Really not a fan of targetting `master` zig, but given the fact you currently are the only consumer of this lib, I won't block you. When 0.10.2 comes out, though, let's use that so that your changes still work and we enjoy some form of stability still.
